src/Entities/Entity.cpp

```diff
@@ -1080,57 +1080,126 @@ void cEntity::HandlePhysics(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	NextSpeed += m_WaterSpeed;
 
-	if (NextSpeed.SqrLength() > 0.0f)
-	{
-		Vector3d HitCoords;
-		Vector3i HitBlockCoords;
-		eBlockFace HitBlockFace;
-		Vector3d wantNextPos = NextPos + NextSpeed * DtSec.count();
-		auto isHit = cLineBlockTracer::FirstSolidHitTrace(*GetWorld(), NextPos, wantNextPos, HitCoords, HitBlockCoords, HitBlockFace);
-		if (isHit)
+	if (NextSpeed.HasNonZeroLength())
+	{
+		Vector3d SizeLimit[]
 		{
-			// Set our position to where the block was hit, minus a bit:
-			// TODO: The real entity's m_Width should be taken into account here
-			NextPos = HitCoords - NextSpeed.NormalizeCopy() * 0.1;
-			if (HitBlockFace == BLOCK_FACE_YP)
-			{
-				// We hit the ground, adjust the position to the top of the block:
-				m_bOnGround = true;
-				NextPos.y = HitBlockCoords.y + 1;
-			}
+			{ m_Width, m_Height, m_Width },
+			{ m_Width, 0, 0 },
+			{ 0, m_Height, 0 },
+			{ 0, 0, m_Width },
+		};
```

Comment:
This just reminds me of a matrix... Maybe create a `cMatrix` class, if you need to do some matrix operations?